src/core/classes/language.ts

```diff
@@ -22,17 +22,17 @@ export default class Language extends EventTarget {
 			this.require.add(this.def.base);
 		}
 		if (this.def.require) {
-			this.require.addAll(this.def.require);
+			this.require.addAll(this.def.require as LanguageProto | readonly LanguageProto[]);
 		}
 
 		for (let def of this.require) {
-			let language = this.registry.peek(def);
+			let language = this.registry.peek(def as LanguageProto);
 			if (language) {
 				// Already resolved
 				this.languages[def.id] = language;
 			}
 			else {
-				this.registry.add(def);
+				this.registry.add(def as LanguageProto);
```

Comment:
But `def` is already defined as `LanguageProto`, why is this needed?

Reply:
Here, `def` has the `LanguageLike` type, so we need to narrow it down.